### DISCUSSION TITLE
Supabase local development

### DISCUSSION THREAD

**Discussion by ANYURU** [supabase/supabase#10975] [Questions]
I have started a new project on supabase and I have some challenges setting up the supabase environment for local development.
Whenever I try to connect the local database to the remote database, I get an error about the supabase local as described below.
On initialising a directory into a supabase project by default the major version is set up to 14. Later it screams that it is incompatible with my remote version when I try to connect the local project to the remote project. I'd like some help in regards to how I can go about this. Thanks. 
```bash
supabase link --project-ref <project ref>
```
I get the output below
```bash
Error: Failed reading config: Invalid db.major_version: 15.
```

**Comment by twfarland** [comment]
Seeing this as well on a fresh project. It's a show stopper.

**Comment by Samurai016** [comment]
### Posting the reply of @silentworks to the duplicated discussion #13476 

<div type='discussions-op-text'>

<sup>Originally posted by **silentworks** April 03, 2023</sup>
This is due to the version in one of your systems being different. Search your `supabase/config.toml` for `major_version` and see what number it's equal to and then in your remote database open the SQL editor and run this query to get the version number.

```sql
select * from
  (select version()) as version,
  (select current_setting('server_version_num')) as version_number;
```

If these numbers are not matching then you will need to either update the version in the `supabase/config.toml` or upgrade your remote database to version 15.
</div>

**Reply by Samurai016** [reply]
@silentworks You got the point, the error is given by just that.

## TL;DR
The remote Postegres database version is 15 or newer but the `supabase\config.toml` supports version 14 at most.
So, if you face this issue, you are unable to link the project to your local environment.

## Steps to reproduce the issue
* You create a new Supabase project, at the moment (April 2023), the database has version 15.1 (`15.1.0.64` to be precise)
* You setup a new local environment with `supabase init` and the generated `supabase\config.toml` contains `major_version = 14`
* You try to link the project with `supabase link` and you get this error message:
  ```bash
  Error: Remote database Postgres version 15 is incompatible with db.major_version 14. If you are setting up a fresh Supabase CLI project, try changing db.major_version in supabase/config.toml to 15.
  ```
* So you update your `config.toml` and set `major_version = 15`
* Retry the `supabase link` command and you get this new error message:
  ```bash
  Error: Failed reading config: Invalid db.major_version: 15.
  ```
**Which is the problem faced from the beginning of this discussion.**
I also tried to set `major_version = 15.1` but I get this message error:
```bash
Error: Failed to read config: toml: line 22 (last key "db.major_version"): incompatible types: TOML value has type float64; destination has type integer
```
So I restored `major_version = 15` but I'm stuck on this.

**Reply by silentworks** [reply]
I think you're probably on an older version of the CLI if the latest major version your `supabase init` adds to the `supabase/config.toml` file is 14. I just did `supabase init` on a new project and the `major_version` is 15 by default. Try upgrading your Supabase CLI.

**Reply by Samurai016** [reply]
It worked, I inadvertently forgot to update the CLI, thanks a lot, that was really a trivial mistake.
